mdpow/workflows/dihedrals.py

```diff
@@ -459,25 +528,24 @@ def dihedral_violins(df, width=0.9, solvents=SOLVENTS_DEFAULT):
                             "interaction",
                             "lambda"]).reset_index(drop=True)
 
-    # number of Coul windows + 1 / number of VDW windows
-    # (+1 for additional space with axes)
     width_ratios = [len(df[df['interaction'] == "Coulomb"]["lambda"].unique()) + 1,
-                    len(df[df['interaction'] == "VDW"]["lambda"].unique())]
+                    len(df[df['interaction'] == "VDW"]["lambda"].unique()),
+                    len(df[df['interaction'] == "Coulomb"]["lambda"].unique()) - 1]
 
-    solvs = np.asarray(solvents)
-    solv2 = 'octanol'
-    if solvs.size > 1:
-        solv2 = solvs[1]
+    solvs = list(solvents)
+    if len(solvs) < 2:
+        solvs.append('N/A')
     
     g = sns.catplot(data=df, x="lambda", y="dihedral", hue="solvent", col="interaction",
                     kind="violin", split=True, width=width, inner=None, cut=0,
                     linewidth=0.5,
-                    hue_order=[solvs[0], solv2], col_order=["Coulomb", "VDW"],
+                    hue_order=[solvs[0], solvs[1]], col_order=["Coulomb", "VDW", "Structure"],
```

Comment:
If `solvents` only contains a single entry then we should have `hue_order=[solvents[0]]`. If it contains two entries we need `hue_order=solvents`. We can get this behavior by just slicing `[:2]`:
```suggestion
                    hue_order=solvents[:2], col_order=["Coulomb", "VDW", "Structure"],
```
Note that we don't need the conversion `solvs = list(solvents)` anymore.

Here's how slicing behaves
```python
In [1]: solvents = ("water", "octanol")

In [2]: solvents[:2]
Out[2]: ('water', 'octanol')

In [3]: solvents[:1]
Out[3]: ('water',)

In [4]: solvents[:1][:2]
Out[4]: ('water',)
```

Reply:
@orbeckst 
Can we manually set the available hues used for single solvent cases so that a plot containing only octane etc. is not blue? Not pertinent, but would be more consistent.. maybe leave alone for now and raise separate issue?

Reply:
probably, but as you said: not high priority, raising an issue would be sufficient